.github/workflows/heartbeat.yml

```diff
@@ -0,0 +1,51 @@
+name: Heartbeat DKG
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 0 * * 1'
+
+jobs:
+  initiate_dkg:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3.12.4'
+
+    - name: Install dependencies
+      run: |
+        sed -i '/^nucypher-core==/d' requirements.txt
+        pip3 install -e . -r requirements.txt
+
+    - name: Import Ape Account
+      run: .github/scripts/import_account.py
+      env:
+        DKG_INITIATOR_PRIVATE_KEY: ${{ secrets.DKG_INITIATOR_PRIVATE_KEY }}
+        DKG_INITIATOR_PASSPHRASE: ${{ secrets.DKG_INITIATOR_PASSPHRASE }}
+
+    - name: Initiate Ritual
+      run: .github/scripts/initiate_dkg.sh
+      env:
+        DKG_INITIATOR_ADDRESS: ${{ secrets.DKG_INITIATOR_ADDRESS }}
+        APE_ACCOUNTS_automation_PASSPHRASE: ${{ secrets.DKG_INITIATOR_PASSPHRASE }}
+        DKG_AUTHORITY_ADDRESS: ${{ secrets.DKG_AUTHORITY_ADDRESS }}
+        RPC_PROVIDER: ${{ secrets.RPC_PROVIDER }}
+        WEB3_INFURA_API_KEY: ${{ secrets.WEB3_INFURA_API_KEY }}
+        POLYGONSCAN_API_KEY: ${{ secrets.POLYGONSCAN_API_KEY }}
+        DOMAIN: ${{ secrets.DOMAIN }}
+        NETWORK: ${{ secrets.NETWORK }}
+        ECOSYSTEM: ${{ secrets.ECOSYSTEM }}
+        ACCESS_CONTROLLER: ${{ secrets.ACCESS_CONTROLLER }}
+        FEE_MODEL: ${{ secrets.FEE_MODEL }}
+
+    - name: Upload Artifact
+      uses: actions/upload-artifact@v3
+      with:
+        name: heartbeat-rituals
+        path: heartbeat-rituals.json
```

Comment:
Does each round of heartbeats overwrite the heartbeat-rituals.json artifact? Just wondering if we happen not to process a heartbeat-rituals.json artifact before another round of heartbeats is performed?

Reply:
It's fine for now, just curious really.

Not sure if we need long-term storage since we will run the reporting script based on rituals in the heartbeat-rituals.json file, and once we do, the heartbeat-rituals.json file is not really needed anymore. Plus I guess we could always determine the rituals that were kicked off from on chain events on the day that the heartbeats were done anyway.